### DISCUSSION TITLE
Designing a System for a Course Management Website - Need Guidance

### DISCUSSION THREAD

**Discussion by hishamarshadc** [community/community#65412] [Programming Help]
I'm currently working on a project for a small startup that offers online courses and installs related hardware in schools. They are now planning to expand their website to accommodate a larger user base. The existing system uses an LMS for providing video content and study materials, but due to leaks, they now want to restrict access to their website. They also need a solution for schools to create or renew custom plans at the end of each educational year.

This is my first major project, and I would greatly appreciate guidance on how to implement this system effectively. How should I structure the user registration, plan creation, and payment process? Are there any specific technologies, tools, or design patterns that would be suitable for this scenario?

Current method I am using.

Onsite inspection and hardware installation are completed. Startup owner manually creates a school account and custom plan. The school receives login credentials and access to the custom plan. Schools use the plan for the educational year.

and my desired approach:

Schools create accounts themselves, set up custom plans, and pay the fees online. The startup owner reviews and approves school requests with minimal interaction.

is there any method that is commonly used in the field to create a plan without account . but when the owner approves it creates an account or something like that. Because i think we will customers if we ask them to create a an account just to enquire about the plans.

**Comment by tuves** [comment]
Welcome to the GitHub Community, @C-hisham, we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category. I've gone ahead and moved it for you. Good luck!

**Reply by hishamarshadc** [reply]
thank you for doing that.
